### PR TITLE
State: Change order of args on editPost and savePost actions

### PR DESCRIPTION
**Pull request by timmyc**
In a few recent PRs @aduth and I have [discussed](https://github.com/Automattic/wp-calypso/pull/7229#discussion_r73565579) how the order of the arguments on the `editPost()` action go against the typical order seen in the other post actions.

In all other actions the argument order is: `siteId`, `postId`.  This branch updates both `editPost()` and `savePost()` to conform to the established argument order of the other post actions.

This branch already touches quite a bit of editor functionality - but seems like it is best that we make this change sooner than later as we are migrating more editor features over to redux.

__To Test__
- Open up a CPT in the editor.
- Validate that changes made to Terms ( hierarchical and token field ) are persisted properly
- Open up a page in the editor
- Confirm that changes to the Order and Parent are persisted properly
- Open up a list view for a CPT, select the draft tab, and hit save from the elipsis menu.  Verify that the post is published properly.

Test live: https://calypso.live/?branch=update/state/posts-editPost

### CODE REVIEW

**Comment by aduth** [comment]
This will need to be rebased to include page templates with the merging of #7229.

File: client/state/posts/test/actions.js [hunk 1]
```diff
@@ -252,7 +252,7 @@ describe( 'actions', () => {
 
 	describe( '#editPost()', () => {
 		it( 'should return an action object for a new post', () => {
-			const action = editPost( {
+			const action = editPost( 2916284, undefined, {
```
Comment (aduth) [inline client/state/posts/test/actions.js]: I think this should work as-is, but one of my only worries with making this change is that `getEditorPostId` will return `null` for a new post, so we'll either need to handle both cases (which I believe we do) or try to normalize via e.g. setting a fallback `null` value for the action creators:

```js
export function savePost( siteId, postId = null, post ) {
```

What do you think?

**Comment by timmyc** [comment]
rebased and added in default of `null` on postId

**Comment by aduth** [comment]
LGTM 👍

**Pull request merged**